src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversion.cs

```diff
@@ -252,7 +252,7 @@ private static void AssertTrivialConversion(ConversionKind kind)
                     break;
             }
 
-            Debug.Assert(isTrivial, "this conversion needs additional data: " + kind);
+            RoslynDebug.Assert(isTrivial, $"this conversion needs additional data: {kind}");
```

Comment:
To emphasize how much these changes matter: this `Debug.Assert` alone was causing 50MB of allocations in the slice I was debugging.